### PR TITLE
feat(nps): update UI

### PR DESCRIPTION
**Pull request by seaerchin**
**NOTE: There's an extra commit because the previous PR was merged into `develop`**
**Please skip `useFeedbackDisclosure`**
**To be reviewed w/ BE PR** ([here](https://github.com/isomerpages/isomercms-backend/pull/863))

## Problem
Previously we were using FormSG as a backend. This updates the nps ui to the one in the figma and hits a backend endpoint.

## Solution
1. add more data to `LoginContext` so we know the `userType` 
2. revamp UI
Screenshot 2023-07-26 at 1 31 44 PM
3. Add calls out to backend to hit our new endpoint 

## Tests
- [ ] go to a markdown page
- [ ] click on back button
- [ ] modal should show up
- [ ] type in some feedback + rating
- [ ] click done
- [ ] modal should close
- [ ] toast should show up

**Notes**
This PR is blocked by the design system upgrade

### CODE REVIEW

**Comment by seaerchin** [comment]
!run-chromatic

File: src/hooks/useFeedbackDisclosure.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  const from = location.state?.from ?? ""
+  // NOTE: We show the feedback modal to the users iff
+  // they are navigating away from the editor
+  const isLeavingContentPage = isEditPageUrl(from) || isSpecialPagesUrl(from)
+  const shouldShowNpsForm = REACT_APP_SHOW_NPS_FORM?.trim() === "true"
+
+  const [lastSeen, setLastSeen] = useFeedbackStorage()
+  // NOTE: Either this is the first time the user has ever seen the survey
+  // or that the user has seen the survey but it has been more than 3 weeks.
+  // Because we toggle the survey on every month, this indicates that they should
```
Comment (kishore03109) [inline src/hooks/useFeedbackDisclosure.ts:59]: hmm shall we be precise on the frequency haha 
7 days vs 3 weeks vs 1 mth

**Review by kishore03109 — APPROVED** [review]
More or less ok with the code, but here is my assumption of the flow here: 

Currently, the `REACT_APP_SHOW_NPS_FORM` is set to true, and the NPS forms are shown once every 3 (or 1 or 4) week. When we set the env var to be false, the timer in their local storage does not reset. Ie, this would mean that when ops/product wants to get feedback, setting the value to `true` does not guarantee that we will be showing NPS to our users. (at the worst case the turnaround time is 4 weeks). as long as ops is aware of this flow this is fine  (and tbh this flow is intuitive since we dont want to spam users with the feedback) 

If this assumption is not correct, please dismiss my review

other nits: if prs are to be reviewed with relevant BE prs, please link them ya! my assumption here is that the relevant be pr is [this](https://github.com/isomerpages/isomercms-backend/pull/863), let me know if this is wrong!

File: .env-example [hunk 2]
```diff
@@ -3,6 +3,7 @@ export REACT_APP_SENTRY_ENV=''
 export REACT_APP_SENTRY_DSN=''
 export REACT_APP_ENV='LOCAL_DEV'
 export ESLINT_NO_DEV_ERRORS=true
+export REACT_APP_SHOW_NPS_FORM=true
```
[the remark this answers is not in the document]
Comment (seaerchin) [reply .env-example:6]: yes

File: src/components/Rating/Rating.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  const paginationStyles = useMultiStyleConfig("Pagination")
+
+  return (
+    <>
+      <ChakraButton
+        variant="unstyled"
+        sx={{
+          // NOTE: We need to use the pagination styles
+          // because the figma uses the pagination button for rating...
+          ...paginationStyles.button,
```
[the remark this answers is not in the document]
Comment (seaerchin) [reply src/components/Rating/Rating.tsx:33]: design system

File: src/hooks/useFeedbackDisclosure.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  // state is NOT guaranteed to exist.
+  const from = location.state?.from ?? ""
+  // NOTE: We show the feedback modal to the users iff
+  // they are navigating away from the editor
+  const isLeavingContentPage = isEditPageUrl(from) || isSpecialPagesUrl(from)
+  const shouldShowNpsForm = REACT_APP_SHOW_NPS_FORM?.trim() === "true"
+
+  const [lastSeen, setLastSeen] = useFeedbackStorage()
+  // NOTE: Either this is the first time the user has ever seen the survey
+  // or that the user has seen the survey but it has been more than 3 weeks.
```
[the remark this answers is not in the document]
Comment (seaerchin) [reply src/hooks/useFeedbackDisclosure.ts:58]: 1 week, will update the commenbt

File: src/hooks/useFeedbackDisclosure.ts (hunk 1 above)
Comment (seaerchin) [reply src/hooks/useFeedbackDisclosure.ts:59]: 2 things here - frequency of survey is 1 month, duration is 1 wek

**Comment by kishore03109** [comment]
!run-chromatic

**Pull request merged**